File: backend/SpringBootLibrary/src/main/java/com/librayapp/springBootLibrary/ResponseModels/CatalogModel.java
package com.librayapp.springBootLibrary.ResponseModels;

import com.librayapp.springBootLibrary.entity.Book;
import lombok.Data;

import java.util.List;

@Data
public class CatalogModel<T> {

    List<T> items;
    long totalItems;
    int totalPages;

    public CatalogModel(List<T> list, long items, int pages) {
        this.items = list;
        this.totalPages = pages;
        this.totalItems = items;
    }
}

File: frontend/react-library/src/Models/CatalogModel.ts
import BookModel from "./BookModel";

export default class CatalogModel<T> {

    items: T[];
    totalItems: number;
    totalPages: number;

    constructor(list: T[], size: number, pages: number) {
        this.items = list;
        this.totalItems = size;
        this.totalPages = pages;
    }
}